packages/app/src/server/models/bookmark-folder.ts

```diff
@@ -0,0 +1,76 @@
+import { Ref, IUser } from '@growi/core';
+import {
+  Types, Document, Model, Schema,
+} from 'mongoose';
+
+
+import loggerFactory from '../../utils/logger';
+import { getOrCreateModel } from '../util/mongoose-utils';
+
+const logger = loggerFactory('growi:models:bookmark-folder');
+
+export type IBookmarkFolderDocument = {
+  name: string
+  owner: Ref<IUser>
+  parent?: string
+}
+export interface BookmarkFolderDocument extends Document {
+  _id: Types.ObjectId
+  name: string
+  owner: Types.ObjectId
+  parent?: Types.ObjectId | null
+}
+
+export interface BookmarkFolderModel extends Model<BookmarkFolderDocument>{
+  createByParameters(params: IBookmarkFolderDocument): IBookmarkFolderDocument
+  findParentFolderByUserId(user: Types.ObjectId | string): IBookmarkFolderDocument[]
+  findChildFolderById(parentBookmarkFolder: Types.ObjectId | string): Promise<IBookmarkFolderDocument[]>
+  deleteFolderAndChildren(bookmarkFolderId: string): void
+  updateBookmarkFolder(bookmarkFolderId: string, name: string, parent: string): BookmarkFolderDocument | null
+}
+
+const bookmarkFolderSchema = new Schema<BookmarkFolderDocument, BookmarkFolderModel>({
+  name: { type: String },
+  owner: { type: Schema.Types.ObjectId, ref: 'User', index: true },
+  parent: { type: Schema.Types.ObjectId, refPath: 'BookmarkFolder', required: false },
+});
+
+
+bookmarkFolderSchema.statics.createByParameters = async function(params: IBookmarkFolderDocument): Promise<BookmarkFolderDocument> {
+  const { name, owner, parent } = params;
+  const parentFolder = await this.findById(parent);
```

Comment:
This method should throw an error when the specified ID of the parent is invalid. Please create a custom error class that extends `ExtensibleCustomError`